src/domains/dashboard/components/Migration/MigrationBanner.tsx

```diff
@@ -8,57 +8,83 @@ import { Image } from "@/app/components/Image";
 import { useLink } from "@/app/hooks/use-link";
 import { ProfilePaths } from "@/router/paths";
 import { migrationGuideUrl, migrationLearnMoreUrl } from "@/utils/polygon-migration";
+import { useMigrations } from "@/app/contexts";
+import { MigrationDisclaimer } from "@/domains/migration/components/MigrationDisclaimer";
 
 export const MigrationBanner = () => {
 	const { t } = useTranslation();
 	const { openExternal } = useLink();
 	const { isDarkMode } = useTheme();
 	const history = useHistory();
 	const profile = useActiveProfile();
+	const { migrations } = useMigrations();
+	const [isDisclaimerOpen, setIsDisclaimerOpen] = useState(false);
 
-	const migrateButtonHandler = useCallback(() => {
-		const path = generatePath(ProfilePaths.Migration, { profileId: profile.id() });
-
+	const confirmHandler = useCallback(() => {
+		const path = generatePath(ProfilePaths.MigrationAdd, { profileId: profile.id() });
 		history.push(path);
 	}, [history, profile]);
 
+	const migrateButtonHandler = useCallback(() => {
+		if (migrations.length > 0) {
+			const path = generatePath(ProfilePaths.Migration, { profileId: profile.id() });
```

Comment:
@goga-m Currently, the transactions are being loaded automatically, the question is whether to show some loading status while transactions are being loaded or just act like the users have no migrations temporarily (show modal). On the first load, it takes a few seconds, after that, it will take like 1 second since data is cached

What do you think? disabling the button temporary?

cc @ItsANameToo

Reply:
@alfonsobries @ItsANameToo what about going with something like:

We always can show the button enabled and when the user clicks the button:
* If migrations are loading, redirect to the migration overview page (just to make sure we don't miss any existing migrations forcing the user to the creation page)
* If migrations are finished loading and there are no migrations go to the creation flow
* If migrations are finished loading and there are migrations, go to migration overview page.

Reply:
@goga-m Im good with that option; only users with no migrations are going to see inconsistencies in edge some cases, lets see what @ItsANameToo thinks